1. Say "Hello, World!" with python

In [14]:
str1 = "Hello, World!"
print(str1)

Hello, World!


2. Given an integer, , perform the following conditional actions:

If n is odd, print Weird
If n is even and in the inclusive range of 2 to 5 , print Not Weird
If n is even and in the inclusive range of 6 to 20 , print Weird
If n is even and greater than 20, print Not Weird

In [37]:
n = int(input())

if n % 2 != 0:
    print("Weird")
elif 2 <= n <= 5:
    print("Not Weird")
elif 6 <= n <= 20:
    print("Weird")
else: 
    print("Not Weird")


 123


Weird


In [40]:
n = int(input())

if n < 1 or n > 100:
    print("Number not in range") 
else: 
    if n%2 != 0:
        print("Weird") 
    elif n%2 == 0 and n>=2 and n<=5: 
        print("Not Weird") 
    elif n%2 == 0 and n>=6 and n<=20: 
        print("Weird") 
    elif n%2 == 0 and n>20: 
        print("Not Weird")

 456


Number not in range


3. The provided code stub reads two integers from STDIN,  a and b. Add code to print three lines where:

The first line contains the sum of the two numbers.
The second line contains the difference of the two numbers (first - second).
The third line contains the product of the two numbers.

In [43]:
a = int(input())
b = int(input())

add = a+b
print(add)

sub = a-b
print(sub)

mul = a*b
print(mul)

 3
 5


8
-2
15


4. The provided code stub reads two integers, a and b, from STDIN.

Add logic to print two lines. The first line should contain the result of integer division,  // . The second line should contain the result of float division,  / .

No rounding or formatting is necessary.

In [44]:
a = int(input())
b = int(input())

integer_div = a//b
print("The result of the integer division is =", integer_div)

fl_div = a/b
print("The result of the float division is =", fl_div)

 3
 5


The result of the integer division is = 0
The result of the float division is = 0.6


In [46]:
# for hackerrank website

a = int(input())
b = int(input())

print(a // b)
print(a / b)


 4
 3


1
1.3333333333333333


5. The provided code stub reads an integer n, from STDIN. For all non-negative integers i<n , print i*i.
the square of each number in seperate line.

In [48]:
n = int(input("enter a number"))

for i in range(0,n):
    i = i*i
    print(i)
    

enter a number 3


0
1
4


6. Given a year, determine whether it is a leap year. If it is a leap year, return the Boolean True, otherwise return False.

Note that the code stub provided reads from STDIN and passes arguments to the is_leap function. It is only necessary to complete the is_leap function.

In [52]:
def is_leap(year):
    leap = False

    if year % 400 == 0:
        leap = True
    elif year % 100 == 0:
        leap = False
    elif year % 4 == 0:
        leap = True
    return leap

year = int(input())
print(is_leap(year))

 1992


True


7. Print the list of integers from 1 to n as a string without spaces
input 3
output 123

In [57]:
n = int(input())
for i in range(1,n+1):
    print(i, end='')

 3


123

8. Let's learn about list comprehensions! You are given three integers x,y,z  and  representing the dimensions of a cuboid along with an integer n. 
Print a list of all possible coordinates given by (i,j,k) on a 3D grid where the sum of i+j+k is not equal to n. 
Here, . Please use list comprehensions rather than multiple loops, as a learning exercise.

Sample Input 
 1
 1
 1
 2
Sample Output
[[0, 0, 0], [0, 0, 1], [0, 1, 0], [1, 0, 0], [1, 1, 1]]

In [59]:
x = int(input())
y = int(input())
z = int(input())
n = int(input())

result = [[i, j, k] 
          for i in range(x + 1) 
          for j in range(y + 1) 
          for k in range(z + 1) 
          if i + j + k != n]

print(result)


 1
 1
 1
 2


[[0, 0, 0], [0, 0, 1], [0, 1, 0], [1, 0, 0], [1, 1, 1]]
